### PR TITLE
Feature/implement respond method

### PR DESCRIPTION
**Pull request by chaak**

### CODE REVIEW

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/WalletConnectScope.kt [hunk 1]
```diff
@@ -39,6 +41,11 @@ internal val keyChain = object : KeyChain {
     }
 }
 
+private val factory: PolymorphicJsonAdapterFactory<EngineData.JsonRpcResponse> =
```
Comment (TalhaAli00) [inline walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/WalletConnectScope.kt:44]: change name to something like `PolymorphicAdapterFactory`

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/client/WalletConnectClientData.kt [hunk 2]
```diff
@@ -33,6 +38,34 @@ sealed class WalletConnectClientData {
...
+            override val id: Long,
+            val error: Error,
+            override val jsonrpc: String = "2.0"
+        ) : JsonRpcResponse()
+
+        data class Error(
+            val code: Long,
+            val message: String,
+        )
+    }
```
Comment (TalhaAli00) [inline walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/client/WalletConnectClientData.kt:70]: Move JsonRpcResponse outside of sealed class if doesn't relate to the sealed class

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/engine/model/EngineData.kt [hunk 3]
```diff
@@ -31,4 +37,28 @@ sealed class EngineData {
...
+            val error: Error,
+            val jsonrpc: String = "2.0"
+        ) : JsonRpcResponse()
+
+        @JsonClass(generateAdapter = true)
+        data class Error(
+            val code: Long,
+            val message: String,
+        )
+    }
```
Comment (TalhaAli00) [inline walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/engine/model/EngineData.kt:63]: Move JsonRpcResponse outside of sealed class if doesn't relate to the sealed class or extend EngineData

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/relay/WakuRelayRepository.kt [hunk 4]
```diff
@@ -46,7 +46,7 @@ class WakuRelayRepository internal constructor(
         Scarlet.Builder()
             .backoffStrategy(LinearBackoffStrategy(TimeUnit.MINUTES.toMillis(DEFAULT_BACKOFF_MINUTES)))
             .webSocketFactory(okHttpClient.newWebSocketFactory(getServerUrl()))
-            .lifecycle(AndroidLifecycle.ofApplicationForeground(application)) // TODO: Maybe have debug version of scarlet w/o application and release version of scarlet w/ application once DI is setup
+//            .lifecycle(AndroidLifecycle.ofApplicationForeground(application)) // TODO: Maybe have debug version of scarlet w/o application and release version of scarlet w/ application once DI is setup
```
Comment (TalhaAli00) [inline walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/relay/WakuRelayRepository.kt:49]: uncomment again to use with sample application

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/relay/WakuRelayRepository.kt [hunk 5]
```diff
@@ -59,7 +59,7 @@ class WakuRelayRepository internal constructor(
     internal val observePublishResponseError: Flow<Relay.Publish.JsonRpcError> = relay.observePublishError()
     internal val observeSubscribeResponse = relay.observeSubscribeAcknowledgement()
     internal val observeUnsubscribeResponse = relay.observeUnsubscribeAcknowledgement()
-    internal val subscriptionRequest: Flow<Relay.Subscription.Request> =
+    internal  val subscriptionRequest: Flow<Relay.Subscription.Request> =
```
Comment (TalhaAli00) [inline walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/relay/WakuRelayRepository.kt:62]: remove extra space

**Review by TalhaAli00 — APPROVED** [review]

File: walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/WalletConnectClient.kt [hunk 6]
```diff
@@ -74,11 +72,27 @@ object WalletConnectClient {
...
                 onFailure = { error -> listener.onError(error) }
             )
         }
     }
+
+    fun respond(
+        responseParams: ClientTypes.ResponseParams,
+        listener: WalletConnectClientListeners.SessionPayload
+    ) = with(responseParams) {
+        val engineResponse = when (jsonRpcResponse) {
```
[the remark this answers is not in the document]
Comment (chaak) [reply walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/WalletConnectClient.kt:87]: Cuz we have a clear separation between the client and engine models. I'm gonna add more descriptive name.

**Pull request merged**